### PR TITLE
Add tests for `cupy.angle()`

### LINKED ISSUE #9296: Dask angle() function + Cupy are reporting errors in different versions
**What happened**:

To reproduce this issue, I'm running a simple code like the one below.

```python
import cupy as cp
import dask.array as da

in_shape = (20, 20, 20)
in_shape_chunks = (5, 5, 5)

rng = cp.random.default_rng(seed=42)
in_data = rng.random(in_shape)

in_data = da.from_array(in_data, chunks=in_shape_chunks)

angle = da.angle(in_data)

np_angle = angle.compute()
```

After executing the code snippet above, I get the following error:

```python
AttributeError: 'cupy._core.core.ndarray' object has no attribute '__array_wrap__'
```

When I move to Dask **2022.7.0**, I get the error below:

```python
TypeError: Cannot convert numpy.bool_ to cupy._core.core.ndarray
```

According some old bugs it is clear that CuPy does not have `__array_wrap__` implemented. Other Dask issues with other functions were fixed, but apparently, `dask.array.angle()` is still buggy.

**What you expected to happen**:

This code should run without any issue. Especially, because it is a simple one. Obviously, it is an error associated more with CuPy.

**Environment**:

- Dask version: 2021.11.2 and 2022.7.0
- Python version: Python 3.8.12
- Operating System: Ubuntu 20.04
- Install method (conda, pip, source): using RAPIDS AI docker image: rapidsai/rapidsai:21.12-cuda11.2-runtime-ubuntu20.04-py3.8

### PR DESCRIPTION
**Pull request by pentschev**
Depends on https://github.com/cupy/cupy/pull/6905 .

- [x] Closes #9296 
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/array/tests/test_cupy_core.py [hunk 1]
```diff
@@ -105,6 +105,22 @@
     lambda x: np.nanprod(x),
     lambda x: np.nanmin(x),
     lambda x: np.nanmax(x),
+    pytest.param(
+        lambda x: np.angle(x),
+        marks=pytest.mark.skipif(
+            cupy_version < parse_version("11.0.0rc2"),
```
Comment (jrbourbeau) [inline dask/array/tests/test_cupy_core.py:111]: It feels a little strange to rely on release candidates when doing version checks. I wonder if a more direct check like `dask.utils.has_keyword(cupy.angle, "deg")` might be more clear.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @pentschev!

File: dask/array/tests/test_cupy_core.py (hunk 1 above)
Comment (pentschev) [reply dask/array/tests/test_cupy_core.py]: That is fine by me as well, done in https://github.com/dask/dask/pull/9312/commits/1fee85ab61e89ece8a9cb16e2e6d56b3f7ab3b28. Thanks James!

**Review by jrbourbeau — APPROVED** [review]
Thanks @pentschev!

Note the non-gpuCI failures are unrelated to the code changes here (xref https://github.com/dask/dask/issues/9323). Merging as these tests are only run in the gpuCI build, which passed as expected

**Comment by pentschev** [comment]
Thanks @jrbourbeau for reviewing and merging!

**Pull request merged**